### PR TITLE
refactor: controller slimming

### PR DESCRIPTION
**Pull request by alvinhui**
## Background

There are a lot of redundant logic in controller now, such as:

- Declaring parameters;
- Catch error;
- Run callback.

We can encapsulate these logic at the entrance, so controller will be more cleaner.

A example for import project:

## Before

```javascript
 async create(ctx) {	
  const { projectManager } = app;
  const { args } = ctx;
  const { path } = args[0];
  const callback = args[args.length - 1];	

  let projects = [];	
  let error;	
  try {	
    projects = await projectManager.createProject(path);	
  } catch (err) {	
    error = err;	
  }	

  callback({	
    error,	
    data: projects	
  });	
}
```

## After

```javascript
async create(ctx) {
  const { args } = ctx;
  const { path } = args;
  const { projectManager } = app;
  return await projectManager.createProject(projectPath);
}
```

Never need to  care for the error, just catch it at the top of the socket.io and agreed with the client how to handle the error.

### CODE REVIEW

File: packages/iceworks-server/src/app/router.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    ['project.index.current', project.index.getCurrent],
+    ['project.index.setCurrent', project.index.setCurrent],
+    ['project.page.list', project.page.list],
+    ['project.dependency.list', project.dependency.list],
+    ['project.dev.start', project.dev.start],
+    ['project.dev.stop', project.dev.stop],
+    ['project.dev.settings', project.dev.settings],
+    ['project.dev.detail', project.dev.detail],
+    ['project.configuration.settings', project.configuration.settings]
+  ];
```
[the remark this answers is not in the document]
Comment (alvinhui) [reply packages/iceworks-server/src/app/router.js]: I think use the same args for `io.route` will be more compatibility:

```javascript
  app.io.route('project.dev.stop', project.dev.stop);
```

[the remark this answers is not in the document]
Comment (chenbin92) [reply packages/iceworks-server/src/app/router.js]: 这不一样吧，你是基于这里抽象了一层，从语义和对象以及数组的使用场景来看，用二维数组会比较不直观吧

[the remark this answers is not in the document]
Comment (chenbin92) [reply packages/iceworks-server/src/app/router.js]: 建议用对象的形式

[the remark this answers is not in the document]
Comment (alvinhui) [reply packages/iceworks-server/src/app/router.js]: 是为了兼容多参数的场景：[route](https://github.com/eggjs/egg-socket.io/blob/master/lib/socket.io/namespace.js#L10)

[the remark this answers is not in the document]
Comment (chenbin92) [reply packages/iceworks-server/src/app/router.js]: 没看懂多参数的使用，给个场景或者例子？

File: packages/iceworks-server/src/app/router.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
```
[the remark this answers is not in the document]
Comment (chenbin92) [reply packages/iceworks-server/src/app/router.js]: 这里改一下?

**Pull request merged**